scripts/core/exp-cooperative-kernels.yml

```diff
@@ -78,6 +78,12 @@ params:
     - type: $x_kernel_handle_t
       name: hKernel
       desc: "[in] handle of the kernel object"
+    - type: size_t
+      name: localWorkSize
+      desc: "[in] number of local work-items that will form a work-group when the kernel is launched"
```

Comment:
This maps directly to the CUDA interface. However the HIP interface expresses the local work size as a three dimensional range (DIM3 in hip parlance): https://rocm.docs.amd.com/projects/HIP/en/latest/doxygen/html/group___module.html#gaa516e011bb07d01550102c98adb57ec2

You get get a 1D work-group size from a 3D range, but not vice versa (or at least the mapping is not 1:1). So it would appear it is more natural that this parameter corresponds to a vector of ints/size_t/3d range etc. This is assuming that the triple is not redundant with regards to behaviour for the hip backend, such that it would make a difference if you e.g. map workgroup size 512 to (16,8,4) vs (8,8,8). I don't know the answer to this question.

I think the same issue applies to the global range parameter.

Reply:
You are right. Sorry my mistake.